### ISSUE TITLE
Add `ByteVector::copyToBuffer(ByteBuffer):Int`

### ISSUE THREAD

**Issue by sirthias** [scodec/scodec-bits#66]
I am currently writing a common abstraction across the several "immutable sequence of bytes" implementations that exist in the Scala library landscape (most importantly `akka.util.ByteString` and `scodec.bits.ByteVector`).

I really like the `ByteVector` API as it's quite rich and a bit more principled than the alternatives, but there appears to be one thing missing that is quite important, this method here:

```scala
/**
   * Copy as many bytes as possible to the given [[ByteBuffer]], starting from its
   * current position. This method will not overflow the buffer.
   *
   * @param buffer a ByteBuffer to copy bytes to
   * @return the number of bytes actually copied
   */
  def copyToBuffer(buffer: ByteBuffer): Int
```

When dealing with streams of `ByteVector` instances that are, for example, encoded/decoded or compressed/decompressed by relying on the JDK's ByteBuffer-based APIs this method is essential for minimizing redundant copying.

**Comment by mpilquist** [comment]
:+1: I'll try to add this sometime this week.

**Comment by sirthias** [comment]
Excellent! Thanks, Michael!

**Comment by mpilquist** [comment]
@sirthias Just published 1.1.3 to Sonatype. Sorry for the delay -- was busy with holiday stuff. :)

**Comment by sirthias** [comment]
Awesome, Michael!
There is nothing to be sorry for! At the contrary, thank you for the quick turn-around!
